# Fix make file to allow CXXFLAGS adjustment, generate and install man page
# Copyright (C) 2017 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# $Id: cppcheck-1.62.diff,v 1.1 2017/10/12 18:36:53 gsotirov Exp $
#
diff -urNad cppcheck-1.62-orig/Makefile cppcheck-1.62/Makefile
--- cppcheck-1.62-orig/Makefile	2013-10-12 13:26:38.000000000 +0300
+++ cppcheck-1.62/Makefile	2017-10-12 21:26:55.509265785 +0300
@@ -55,6 +55,8 @@
 
 ifndef CXXFLAGS
     CXXFLAGS=-O2 -DNDEBUG -Wall
+else
+    CXXFLAGS+=-O2 -DNDEBUG -Wall
 endif
 
 ifeq ($(HAVE_RULES),yes)
@@ -87,10 +89,11 @@
 endif
 
 BIN=$(DESTDIR)$(PREFIX)/bin
+MAN=$(DESTDIR)$(PREFIX)/man
 
 # For 'make man': sudo apt-get install xsltproc docbook-xsl docbook-xml on Linux
-DB2MAN=/usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
-XP=xsltproc -''-nonet -''-param man.charmap.use.subset "0"
+DB2MAN=/etc/asciidoc/docbook-xsl/manpage.xsl
+XP=xsltproc -''-nonet -''-param man.charmap.use.subset "0" --output man/
 MAN_SOURCE=man/cppcheck.1.xml
 
 
@@ -202,7 +205,7 @@
 cppcheck: $(LIBOBJ) $(CLIOBJ) $(EXTOBJ)
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -o cppcheck $(CLIOBJ) $(LIBOBJ) $(EXTOBJ) $(LIBS) $(LDFLAGS)
 
-all:	cppcheck testrunner
+all:	cppcheck testrunner man
 
 testrunner: $(TESTOBJ) $(LIBOBJ) $(EXTOBJ) cli/threadexecutor.o cli/cmdlineparser.o cli/cppcheckexecutor.o cli/filelister.o cli/pathmatch.o
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -o testrunner $(TESTOBJ) $(LIBOBJ) cli/threadexecutor.o cli/cppcheckexecutor.o cli/cmdlineparser.o cli/filelister.o cli/pathmatch.o $(EXTOBJ) $(LIBS) $(LDFLAGS)
@@ -231,10 +234,12 @@
 tags:
 	ctags -R --exclude=doxyoutput .
 
-install: cppcheck
+install: cppcheck man/cppcheck.1
 	install -d ${BIN}
 	install cppcheck ${BIN}
 	install htmlreport/cppcheck-htmlreport ${BIN}
+	install -d ${MAN}/man1
+	install man/cppcheck.1 ${MAN}/man1/cppcheck.1
 
 
 ###### Build
